homeassistant/components/sensor/whois.py

```diff
@@ -121,16 +109,30 @@ def update(self):
             return
 
         if response:
-            self._data = response
+            if 'expiration_date' not in response:
+                _LOGGER.error(
+                    "Failed to find expiration_date in whois lookup response. "
+                    "Did find: %s", ', '.join(response.keys()))
```

Comment:
Clear out `self._state` and `self._attributes` if we get an invalid response.

Reply:
Added another class method to clear the state / attributes on error as I've got a few cases already.